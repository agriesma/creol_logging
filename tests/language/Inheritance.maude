load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "A" : Class | Inh: noInh, Param: "a", Att: "a" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
      Ocnt: 0 >
    
    < "B" : Class | Inh: "A" < "b" >, Param: "b", Att: "b" |-> null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assert
              "=" ( "a" :: "b" ) ; return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
      Ocnt: 0 > } .
endm

