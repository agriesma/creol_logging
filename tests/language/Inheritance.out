==========================================
reduce in PROGRAM : classes .
result Configuration: 
< "A" : Class | Inh: noInh, Param: "a", Att: "a" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(
    emp) >,
  Ocnt: 0 >
 
< "B" : Class | Inh: "A" < "b" >, Param: "b", Att: "b" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assert "="("a" ::
    "b") ; return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null,
    Code: skip ; return(emp) >,
  Ocnt: 0 >

==========================================
rewrite in PROGRAM : main(classes, "B", int(1)) .
result State: {
< ob("B0") : "B" |
  Att: "a" |-> int(1), "b" |-> int(1), "this" |-> ob("B0"),
  Pr: idle,
  PrQ: noProc,
  Lcnt: 3 >
 
< ob("main") : "" |
  Att: noSubst,
  Pr: idle,
  PrQ: noProc,
  Lcnt: 0 >
 
< "A" : Class | Inh: noInh, Param: "a", Att: "a" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp)
    >, < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(
    emp) >,
  Ocnt: 0 >
 
< "B" : Class | Inh: "A" < "b" >, Param: "b", Att: "b" |-> null, Mtds:
  < "init" : Method | Param: noVid, Att: "_" |-> null, Code: assert "="("a" ::
    "b") ; return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null,
    Code: skip ; return(emp) >,
  Ocnt: 1 >
}
Bye.
